### ISSUE TITLE
Destructuring Array assignment with type annotation creates invalid syntax

### ISSUE THREAD

**Issue by chrispaterson** [coderaiser/putout#56]
The code I'm working with has a type annotation and that gets carried over to the destructured array assignment.  This:

```typescript
const nextProps: DialogProps = dialogProps[0];
```

Becomes:
```typescript
const [nextProps: DialogProps] = dialogProps;
```

This is not valid syntax and breaks the TypeScript compiler.  It would probably be best to strip the type annotation and let TypeScript infer the type.  There is no guarantee that `dialogProps` has a length > 0, so the initial code is bad form anyway because the type annotation acts as an assertion as well in this case.

**Comment by coderaiser** [comment]
Thanks a lot :). Yes, I see type should definitely be removed:

![image](https://user-images.githubusercontent.com/1573141/115721631-e520a880-a386-11eb-8f42-6e61552c9e71.png)

Landed in `@putout/plugin-apply-destructuring@5.2.0` 🎉 .
Is it works for you?

**Comment by coderaiser** [comment]
Closed due to a long time of inactivity 🏝
